x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/alert_context_menu.test.tsx

```diff
@@ -131,4 +131,85 @@ describe('InvestigateInResolverAction', () => {
     expect(wrapper.find(markAsAcknowledgedButton).first().exists()).toEqual(true);
     expect(wrapper.find(markAsClosedButton).first().exists()).toEqual(true);
   });
+
+  test('it disables AddEndpointEventFilter when timeline id is not host events page', () => {
+    (useUserPrivileges as jest.Mock).mockReturnValue({
```

Comment:
Thanks for adding these tests!

nit: since 3 / 4 of the tests have the same `customProps`, and 3 / 4 tests also have the same `mockReturnValue`, consider something like the following to eliminate the repeated props / mocking:

```ts
  describe('disabling AddEndpointEventFilter', () => {
    const endpointEventProps = {
      ...props,
      ecsRowData: { ...ecsRowData, agent: { type: ['endpoint'] }, event: { kind: ['event'] } },
    };

    describe('when users can access endpoint management', () => {
      beforeEach(() => {
        (useUserPrivileges as jest.Mock).mockReturnValue({
          ...mockInitialUserPrivilegesState(),
          endpointPrivileges: { loading: false, canAccessEndpointManagement: true },
        });
      });

      test('it disables AddEndpointEventFilter when timeline id is not host events page', () => {
        const wrapper = mount(
          <AlertContextMenu {...endpointEventProps} timelineId={TimelineId.active} />,
          {
            wrappingComponent: TestProviders,
          }
        );

        wrapper.find(actionMenuButton).simulate('click');
        expect(wrapper.find(addEndpointEventFilterButton).first().exists()).toEqual(true);
        expect(wrapper.find(addEndpointEventFilterButton).first().props().disabled).toEqual(true);
      });

      test('it enables AddEndpointEventFilter when timeline id is host events page', () => {
        const wrapper = mount(
          <AlertContextMenu {...endpointEventProps} timelineId={TimelineId.hostsPageEvents} />,
          {
            wrappingComponent: TestProviders,
          }
        );

        wrapper.find(actionMenuButton).simulate('click');
        expect(wrapper.find(addEndpointEventFilterButton).first().exists()).toEqual(true);
        expect(wrapper.find(addEndpointEventFilterButton).first().props().disabled).toEqual(false);
      });

      test('it disables AddEndpointEventFilter when timeline id is host events page but is not from endpoint', () => {
        const customProps = {
          ...props,
          ecsRowData: { ...ecsRowData, agent: { type: ['other'] }, event: { kind: ['event'] } },
        };
        const wrapper = mount(
          <AlertContextMenu {...customProps} timelineId={TimelineId.hostsPageEvents} />,
          {
            wrappingComponent: TestProviders,
          }
        );

        wrapper.find(actionMenuButton).simulate('click');
        expect(wrapper.find(addEndpointEventFilterButton).first().exists()).toEqual(true);
        expect(wrapper.find(addEndpointEventFilterButton).first().props().disabled).toEqual(true);
      });
    });

    describe('when users can NOT access endpoint management', () => {
      beforeEach(() => {
        (useUserPrivileges as jest.Mock).mockReturnValue({
          ...mockInitialUserPrivilegesState(),
          endpointPrivileges: { loading: false, canAccessEndpointManagement: false },
        });
      });

      test('it disables AddEndpointEventFilter when timeline id is host events page but cannot access endpoint management', () => {
        const wrapper = mount(
          <AlertContextMenu {...endpointEventProps} timelineId={TimelineId.hostsPageEvents} />,
          {
            wrappingComponent: TestProviders,
          }
        );

        wrapper.find(actionMenuButton).simulate('click');
        expect(wrapper.find(addEndpointEventFilterButton).first().exists()).toEqual(true);
        expect(wrapper.find(addEndpointEventFilterButton).first().props().disabled).toEqual(true);
      });
    });
```

Reply:
Will add another top level `describe` for this test case `it enables AddEndpointEventFilter when timeline id is host events page` since is for when `AddEndpointEventFilter` is enabled.